package/MDAnalysis/analysis/density.py

```diff
@@ -372,12 +372,12 @@ def _check_set_unit(self, u):
                 try:
                     units.conversion_factor[unit_type][value]
                     self.units[unit_type] = value
-                except KeyError:
-                    raise ValueError('Unit ' + str(value) + ' of type ' + str(unit_type) + ' is not recognized.')
-        except AttributeError:
+                except KeyError as e:
```

Comment:
We prefer variable names with multiple letters such as `err` as it is a lot easier to search for. (`err` is also already being used elsewhere, as you saw.)

(This comment applies throughout.)